components/shared/embedder/lib.rs

```diff
@@ -433,3 +437,114 @@ pub struct DualRumbleEffectParams {
 pub enum GamepadHapticEffectType {
     DualRumble(DualRumbleEffectParams),
 }
+
+#[derive(Clone, Debug, Deserialize, MallocSizeOf, Serialize)]
+pub struct WebResourceRequest {
+    #[serde(
+        deserialize_with = "::hyper_serde::deserialize",
+        serialize_with = "::hyper_serde::serialize"
+    )]
+    #[ignore_malloc_size_of = "Defined in hyper"]
+    pub method: Method,
+    pub url: ServoUrl,
+    #[serde(
+        deserialize_with = "::hyper_serde::deserialize",
+        serialize_with = "::hyper_serde::serialize"
+    )]
+    #[ignore_malloc_size_of = "Defined in hyper"]
+    pub headers: HeaderMap,
+    pub is_redirect: bool,
+    pub is_main_frame: bool,
+}
+
+impl WebResourceRequest {
+    pub fn new(url: ServoUrl) -> WebResourceRequest {
```

Comment:
When there's only one place that constructs WebResourceRequest values, builder methods like this don't seem like they're providing much value over multiple arguments to the `new` method.